rfc/rfc-0000-match-params.md

```diff
@@ -0,0 +1,138 @@
+# Meta
+[meta]: #meta
+- Name: Match params in field selector
+- Start Date: 2022-02-14
+- Author(s): @jwntrs
+- Status: Draft
+- RFC Pull Request:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The [template switching RFC](https://github.com/vmware-tanzu/cartographer/pull/75) introduced the ability to match templates based on fields in the workload. However matching on params is not well defined. It is currently not possible to match on a field in the supply chain, and matching on fields in the workload is complex. You would need to define something as follows:
```

Comment:
Is the idea here to be able to match on a field (param) in the supply chain that may or may not be overwritten by a workload?

If not, there would be no reason to match on a field in the supply chain, right? Because that field will always be the same.

Reply:
Yeah the idea is to match on the resulting param after we process the [param hierarchy](https://cartographer.sh/docs/v0.2.0/architecture/#parameter-hierarchy). 

This means matching on a supply chain `default` or `value`.

In the cases where the supply chain sets a value (which cannot be overridden) it may still be useful to be able to switch templates on this static value. You might want to maintain a single supply chain, and use ytt to template out the value during installation. This will let you use first class template switching with very minimal ytt logic.